File: src/minecraft/monnef/jaffas/food/block/BlockJaffas.java
/*
 * Jaffas and more!
 * author: monnef
 */

package monnef.jaffas.food.block;

import monnef.core.base.BlockMonnefCore;
import monnef.jaffas.food.JaffasFood;
import monnef.jaffas.food.common.Reference;
import net.minecraft.block.Block;
import net.minecraft.block.material.Material;

public abstract class BlockJaffas extends BlockMonnefCore {
    public BlockJaffas(int id, int texture, Material material) {
        super(id, texture, material);
        setCreativeTab(JaffasFood.CreativeTab);
    }

    @Override
    public String getCustomIconName() {
        return null;
    }

    @Override
    public String getModName() {
        return Reference.ModName;
    }

    @Override
    public int getDefaultSheetNumber() {
        return 1;
    }

    @Override
    public Block setUnlocalizedName(String par1Str) {
        return super.setUnlocalizedName("jaffas." + par1Str);
    }
}

File: src/minecraft/monnef/jaffas/food/block/BlockJaffas.scala
package monnef.jaffas.food.block

import monnef.core.block.BlockMonnefCore
import monnef.jaffas.food.JaffasFood
import net.minecraft.block.Block
import net.minecraft.block.material.Material
import monnef.jaffas.food.common.IconDescriptorJaffas


class BlockJaffas(_id: Int, _texture: Int, _material: Material) extends BlockMonnefCore(_id, _texture, _material) with IconDescriptorJaffas {
  setCreativeTab(JaffasFood.instance.creativeTab)

  override def setUnlocalizedName(par1Str: String): Block = super.setUnlocalizedName("jaffas." + par1Str)
}
